src/io/flutter/jxbrowser/EmbeddedBrowser.java

```diff
@@ -16,25 +16,46 @@
 import com.teamdev.jxbrowser.navigation.event.LoadFinished;
 import com.teamdev.jxbrowser.view.swing.BrowserView;
 import icons.FlutterIcons;
+import org.jetbrains.annotations.NotNull;
 
 import java.awt.*;
+import java.io.File;
+import java.nio.file.Paths;
 import java.util.concurrent.atomic.AtomicBoolean;
 
 import static com.teamdev.jxbrowser.engine.RenderingMode.HARDWARE_ACCELERATED;
 
 public class EmbeddedBrowser {
-  public void openPanel(ContentManager contentManager, String tabName, String url) {
+  private static EmbeddedBrowser embeddedBrowser;
+
+  @NotNull
+  public static EmbeddedBrowser getInstance() {
```

Comment:
Do we want this to be an application singleton or a per project (per IntelliJ window) singleton? I'm guessing the 2nd one. There are several places in the code base where we have per-project singletons. You register them in the plugin's xml file and then have a slightly different singleton getter. See for example `FlutterReloadManager.getInstance(Project project)`.

Reply:
Ack forgot about this! Changed to per-project.